packages/@aws-cdk/cloud-assembly-schema/README.md

```diff
@@ -0,0 +1,62 @@
+## Cloud Assembly Schema
+<!--BEGIN STABILITY BANNER-->
+
+---
+
+![Stability: Stable](https://img.shields.io/badge/stability-Stable-success.svg?style=for-the-badge)
+
+
+---
+<!--END STABILITY BANNER-->
+
+This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project.
+
+## Cloud Assembly
+
+The *Cloud Assembly* is the output of the synthesis operation. It is produced as part of the
+[`cdk synth`](https://github.com/aws/aws-cdk/tree/master/packages/aws-cdk#cdk-synthesize)
+command, or the [`app.synth()`](https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/core/lib/app.ts#L135) method invocation.
+
+Its essentially a set of files and directories, one of which is the `manifest.json` file. It defines the set of instructions that are
+needed in order to deploy the assembly directory.
+
+> For example, when `cdk deploy` is executed, the CLI reads this file and performs its instructions:
+> - Build container images.
+> - Upload assets.
+> - Deploy CloudFormation templates.
+
+Therefore, the assembly is how the CDK class library and CDK CLI (or any other consumer) communicate. To ensure compatibility
+between the assembly and its consumers, we treat the manifest file as a well defined, versioned schema.
+
+## Schema
+
+This module contains the typescript structs that comprise the `manifest.json` file, as well as the
+generated [*json-schema*](./schema/cloud-assembly.schema.json).
+
+## Versioning
+
+The schema version is specified in the [`cloud-assembly.version.json`](./schema/cloud-assembly.schema.json) file, under the `version` property.
+It follows semantic versioning, but with a small twist.
```

Comment:
Well, I would argue that it is semantic versioning, except that we determine what is considered a `major` bump. 

We still adhere to semantic versioning by only introducing breaking changes with `major` version bumps, except that they happen even in scenarios that normal schema versioning would consider non-breaking. 

Semantic versioning doesn't necessarily define what "breaking" means, its just a form of communication, so as long as we keep this communication honest, i.e, bumping the `major` version every time we introduce a change that we consider is breaking, we are ok. 

I actually like that we use the `semantic` terminology because it reminds people to be aware of `major` versions bumps.